airflow/providers/ssh/hooks/ssh.py

```diff
@@ -101,7 +103,8 @@ def __init__(
         password: Optional[str] = None,
         key_file: Optional[str] = None,
         port: Optional[int] = None,
-        timeout: int = 10,
+        timeout: Optional[int] = 10,
```

Comment:
Is this change needed/correct? I feel we want to either change the default to `None` (so we can better detect whether `timeout` is passed in by the user), or the type should continue to be `int` (since the user shouldn’t do `timeout=None` in any situation).

Reply:
Yeah, looking at it again you're right. It should default to None otherwise the user will need to purposefully pass timeout = None to turn off the deprecation warning.

In order to maintain backwards compatibility, since the current versions of the hook/operator allow the user to omit timeout, I'll need to handle a situation where no timeout is passed and set the default later in the code.